src/document.ts

```diff
@@ -63,6 +64,69 @@ export class Document {
     return `[Document: ${this.path} (${this.locale.id})]`;
   }
 
+  /**
+   * Lists documents using glob patterns, as outlined by the [`glob`
+   * module](https://github.com/isaacs/node-glob#glob-primer).
+   *
+   * Note the following behavior:
+   * - Files prefixed with `_` are ignored.
+   * - Only files with supported doc extensions are returned.
+   *
+   * Various techniques can be used to list docs depending on your needs:
+   *
+   * ```
+   * // All docs within the "pages" collection:
+   * Document.list(pod, '/content/pages/**')
+   *
+   * // Only Markdown docs within the "pages" collection:
+   * Document.list(pod, '/content/pages/*.md')
+   *
+   * // All docs within both the "pages" and "posts" collections:
+   * Document.list(pod, ['/content/pages/**', '/content/posts/**'])
+   *
+   * // All Markdown docs within the entire pod:
+   * Document.list(pod, '*.md')
+   *
+   * // All docs named `index.yaml` within the entire pod:
+   * Document.list(pod, 'index.yaml')
+   * ```
+   * @param pod The pod object.
+   * @param patterns A list of glob patterns or a single glob pattern. If
+   * nothing is supplied, all docs within the pod will be returned.
+   */
+  static list(pod: Pod, patterns?: string[] | string) {
+    let paths: string[] = [];
+    if (typeof patterns === 'string') {
+      patterns = [patterns];
+    }
+    patterns = patterns || [`${Pod.DefaultContentPodPath}**/*`];
+    patterns.forEach(pattern => {
+      paths = paths.concat(
+        glob.sync(pattern, {
+          cwd: pod.root,
+          root: pod.root,
+          ignore: '/**/*/_*',
+          matchBase: true,
```

Comment:
Do we want to use the `matchBase`? I can see how it would make it 'easier' in some simple searches (like finding all `index.yaml` files, but it would make it harder to find the files only in the current directory.

For example, how would you search for yaml files only in the main directory of the collection? The collection `.docs` would do something like `/content/pages/**` automatically to filter down to the collection, but if you have an argument only get `.yaml` in the main collection directory it would have to be something like `/content/pages/*.yaml` instead of just being able to do `*.yaml` to get only the main level yaml files.